### PR TITLE
Add flush API for deleting all vectors

### PR DESCRIPTION
**Pull request by aknishid**
### Description:

Flush API to delete all vectors has been added.
Previous Related PR → https://github.com/vdaas/vald/pull/1895

### Versions:

- Go Version: 1.21.5
- Docker Version: 20.10.8
- Kubernetes Version: v1.28.4
- NGT Version: 2.1.6

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

### CODE REVIEW

File: pkg/agent/core/ngt/service/ngt.go [hunk 1]
```diff
@@ -1615,6 +1753,11 @@ func (n *ngt) IsIndexing() bool {
 	return i && ok
 }
 
+func (n *ngt) IsFlushing() bool {
+	i, ok := n.flushing.Load().(bool)
```
Comment (kpango) [inline pkg/agent/core/ngt/service/ngt.go:1757]: it would be better to use atomic.Bool

Comment (aknishid) [reply pkg/agent/core/ngt/service/ngt.go]: I've implemented the change, which has simplified the code by removing the need for type assertions when loading the boolean value. 
 https://github.com/vdaas/vald/pull/2273/commits/bc2fe50c40cfecaee1b7aca1edef7847e5e58b94

File: apis/proto/v1/payload/payload.proto [hunk 2]
```diff
@@ -335,6 +335,10 @@ message Remove {
   }
 }
 
+message Flush {
```
Comment (vankichi) [inline apis/proto/v1/payload/payload.proto:338]:
```suggestion
// Flush related messages.
message Flush {
```

**Review by vankichi — CHANGES_REQUESTED** [review]

**Review by vankichi — APPROVED** [review]

**Review by datelier — APPROVED** [review]

**Review by kpango — APPROVED** [review]
LGTM

**Comment by kpango** [comment]
@all-contributors please add @aknishid for code maintenance

**Pull request merged**